src/libcrun/container.c

```diff
@@ -3883,6 +3883,77 @@ libcrun_container_exec_with_options (libcrun_context_t *context, const char *id,
   return ret;
 }
 
+static int
+xstrcmp (const char *l, const char *r)
+{
+  if (l == NULL && r == NULL)
+    return 0;
+  if (l == NULL)
+    return -1;
+  if (r == NULL)
+    return 1;
+
+  return strcmp (l, r);
+}
+
+static int
+dedup_devices (runtime_spec_schema_config_linux_resources *orig_res,
+               runtime_spec_schema_config_linux_resources *res,
+               const char *state_root, const char *id,
+               libcrun_error_t *err)
+{
+  cleanup_free char *marker_file = NULL;
+  size_t i;
+  int ret, exists;
+
+  if (res == NULL || res->devices == NULL)
+    return 0;
+
+  ret = append_paths (&marker_file, err, state_root, id, "devices-modified", NULL);
```

Comment:
state_root can be NULL at this point, so you'd first need to call `libcrun_get_state_directory` to get the container private directory

Reply:
This commit is now removed